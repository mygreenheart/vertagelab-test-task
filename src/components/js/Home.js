import React, { useState } from 'react'
import ItemGradient from "./ItemGradient"
import "../css/style.css"

export default function Home() {
    const [gradientData, setGradientData] = useState(JSON.parse(localStorage.getItem("gradientData")))

    const handleDelete = (id) => {
        const fillterGradientData = gradientData.filter(e => {
            return e.id !== id
        })
        setGradientData(fillterGradientData)
        localStorage.setItem("gradientData", JSON.stringify(fillterGradientData))
    }

    return (
        <div className="gradient-items">
            {(!gradientData) ?
                null :
                gradientData.map(e => {
                    return <ItemGradient key={e.id} id={e.id} colorFrom={e.colorFrom} colorTo={e.colorTo} handleDelete={handleDelete} />
                })}
        </div>
    )
}
