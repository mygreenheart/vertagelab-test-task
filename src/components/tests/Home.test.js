import React from 'react';
import Home from "../js/Home"

import Enzyme, { shallow, mount, render } from 'enzyme';

import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })
const gradientData = [{ "id": "adf64f42-c503-4aae-8017-01dc9a29fa64", "colorFrom": "#ffffff", "colorTo": "#000000" },
{ "id": "4b4b40e3-d253-4e07-9f76-09a7fce192b4", "colorFrom": "#aaaaaa", "colorTo": "#cccccc" },
{ "id": "ab451464-f3e6-44dd-9251-63469043c6c6", "colorFrom": "#abcabc", "colorTo": "#333333" }]
describe('Home js', () => {
    it("shoude correct render", () => {
        const wrapper = shallow(<Home />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })



});

