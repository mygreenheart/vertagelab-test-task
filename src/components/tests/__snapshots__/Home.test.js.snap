// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home js shoude correct render 1`] = `
<div
  className="gradient-items"
/>
`;
